### PR TITLE
#301 mutation testing

### PR DESCRIPTION
**Pull request by slskiba**
Suggested realization of #301 using https://pitest.org/

In IDEasy project directory, run the following command to generate a Pit Test Coverage Report of all of the cli module:
```
mvn test-compile org.pitest:pitest-maven:mutationCoverage -DtargetClasses=com.devonfw.tools.ide* -DtargetTests=com.devonfw.tools.ide* -pl cli
```
To run the mutation test on all of the cli module takes a bit more than an hour to run. To test only a subset of classes, specify them accordingly as targetClasses/targetTests, where targetClasses refer to be classes whose code is to be mutated, and targetTests refers to the tests that the mutations are tested against. Ideally, these two should stand in relation to one another. 

As of 25.07.2024, the report yields the following overall result: 
image
where
**Line Coverage** denotes the amount lines covered by tests in proportion to the total number of executable lines in a class. 
**Mutation Coverage** denotes the amount of mutations killed in proportion to the total number of mutations created (including those not covered by tests). It reaches 100% iff both line coverage and test strength (see below) reach 100%.
**Test Strength** denotes the amount of mutations killed in proportion to the total number of mutations covered by tests

In the report, individual classes can be examined for their test and mutation coverage and the applied mutations. 

A mutation is said to survive if all tests pass with the modified code. If a mutation is killed, it means that at least one test did not pass when run on the mutated code.

### CODE REVIEW

**Comment by jan-vcapgemini** [comment]
Please add: https://github.com/pitest/pitest-junit5-plugin/blob/master/LICENSE.txt to your `LICENSE.txt`

File: pom.xml [hunk 1]
```diff
@@ -84,6 +84,18 @@
...
       </plugin>
+      <plugin>
+        <groupId>org.pitest</groupId>
+        <artifactId>pitest-maven</artifactId>
+        <version>1.16.1</version>
+        <dependencies>
+          <dependency>
+            <groupId>org.pitest</groupId>
+            <artifactId>pitest-junit5-plugin</artifactId>
+            <version>1.2.1</version>
```
Comment (jan-vcapgemini) [inline pom.xml:95]: Shouldn't we add the LICENSE to our documentation too?

**Review by jan-vcapgemini — APPROVED** [review]
Looks good to me. I've added a question about the "missing" LICENSE.

File: pom.xml (hunk 1 above)
Comment (jan-vcapgemini) [reply pom.xml:95]: License is not relevant to users, so this is fixed.

File: .github/workflows/pitest.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Pitest
```
Comment (hohwille) [inline .github/workflows/pitest.yml:1]: Workflow name and filename is a little bit cryptic.
`mutation-test` would be more self explanatory.

**Review by hohwille — APPROVED** [review]
@slskiba thanks for analyzing this innovative topic and providing this solution 👍 
I left a nice to have improvement comment.
Otherwise ready for merge.

**Review by hohwille — APPROVED** [review]
@slskiba thanks for making this perfect. 👍 
Ready for merge now.

**Pull request merged**